### PR TITLE
German credit data use case

### PR DESCRIPTION
**Pull request by pkopper**

### CODE REVIEW

**Comment by pat-s** [comment]
Thanks! Please fix Travis - review is on my list :)

**Comment by mllg** [comment]
The German credit data set is already included in mlr3 (`tsk("german_credit")`). I like the description of the data set though, could you create a PR for this in mlr3? - you might need to adept the variable names a bit.

File: bookdown/09-use-cases.Rmd [hunk 1]
```diff
@@ -350,3 +350,537 @@ res = resample(task = tsk_ftsel$clone()$filter(train.idx), lrn_ranger, cv3)
...
+msr_list = list(msr("classif.ce"), msr("classif.auc"))
+performances = bmr$aggregate(msr_list)
+cbind(learner = names(lrn_list), performances[, c("classif.ce", "classif.auc")])
+```
+
+All learners seem to have similar performance while the `learner_rf_low` seems to underfit a bit.
+
+
+=======
+>>>>>>> 4c806162b1581e77c3294d99c3c646d3f94a6ac6
```
Comment (pat-s) [inline bookdown/09-use-cases.Rmd]: Be aware of this ;)

File: bookdown/09-use-cases.Rmd [hunk 2]
```diff
@@ -350,3 +350,533 @@ res = resample(task = tsk_ftsel$clone()$filter(train.idx), lrn_ranger, cv3)
...
+```{r 09-use-cases-2-023}
+learner_logreg$predict_type = "prob"
+```
+
+```{r 09-use-cases-2-024}
+learner_logreg$predict_newdata(german_credit)
+```
+
+Note that sometimes one needs to be cautious when dealing with the probability interpretation of the predictions.
+<!--For a discussion on this topic, have a look at this article:-->
```
Comment (pat-s) [inline bookdown/09-use-cases.Rmd]: ?

File: bookdown/09-use-cases.Rmd [hunk 3]
```diff
@@ -350,3 +350,533 @@ res = resample(task = tsk_ftsel$clone()$filter(train.idx), lrn_ranger, cv3)
...
+```
+Here, we use "holdout", a simple train-test split (with just one interation).
+We use the `resample()` function to undertake the resampling calculation:
+```{r 09-use-cases-2-026}
+res = resample(task, learner_logreg, resampling)
+res
+```
+The default score of the measure is included in the `$aggregate()` slot:
+```{r 09-use-cases-2-027}
+res$aggregate()
```
Comment (pat-s) [inline bookdown/09-use-cases.Rmd]: watch spacing

File: bookdown/09-use-cases.Rmd [hunk 4]
```diff
@@ -350,3 +350,533 @@ res = resample(task = tsk_ftsel$clone()$filter(train.idx), lrn_ranger, cv3)
...
+If we have prior knowledge on how the learner should be (hyper-)parametrised, the way to go would be manually entering the parameters in the parameter set.
+In most cases, however, we would want to tune the model so that it can search good model configurations itself.
+For now however we only want to compare a small number of models.
+
+To get an idea on which parameters can be manipulated, we can investigate the parameters of the original package version.
+
+```{r 09-use-cases-2-039}
+
+?ranger::ranger
+
```
Comment (pat-s) [inline bookdown/09-use-cases.Rmd]: watch spacing

File: bookdown/09-use-cases.Rmd [hunk 5]
```diff
@@ -350,3 +350,533 @@ res = resample(task = tsk_ftsel$clone()$filter(train.idx), lrn_ranger, cv3)
...
+```
+
+```{r 09-use-cases-2-040, echo= FALSE}
+
+expression1 = floor(sqrt(ncol(german_credit) - 1))
+
```
Comment (pat-s) [inline bookdown/09-use-cases.Rmd]: watch spacing

File: bookdown/09-use-cases.Rmd [hunk 6]
```diff
@@ -350,3 +350,533 @@ res = resample(task = tsk_ftsel$clone()$filter(train.idx), lrn_ranger, cv3)
...
+learner_rf_medium = lrn("classif.ranger", predict_type = "prob")
+
+learner_rf_low = lrn("classif.ranger", predict_type = "prob")
+learner_rf_low$param_set$values <- list(num.trees = 5, mtry = 2)
+
+learner_rf_high = lrn("classif.ranger", predict_type = "prob")
+learner_rf_high$param_set$values <- list(num.trees = 1000, mtry = 11)
+
+
```
Comment (pat-s) [inline bookdown/09-use-cases.Rmd]: watch spacing

File: bookdown/09-use-cases.Rmd (hunk 2 above)
Comment (pkopper) [reply bookdown/09-use-cases.Rmd]: Removed that

**Comment by mllg** [comment]
This fits better into the gallery. You already have a stand-alone Rnw, so copying this PR over mlr3gallery should be straight forward. Let me know if you need help.

**Comment by pfistfl** [comment]
@pkopper:
Can we merge this into the gallery ASAP? Please contact me if you need help, but let's get this done soon-ish.

**Comment by be-marc** [comment]
Included in the gallery.

**Pull request closed without merging**